spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/TaskServiceUtils.java

```diff
@@ -398,4 +399,28 @@ static boolean isUseUserAccessToken(TaskConfigurationProperties taskConfiguratio
 
 		return taskConfigurationProperties.isUseUserAccessToken();
 	}
+
+	/**
+	 * Move and configure app arguments to the composed-task-app-arguments argument.  Also convert app args to base64 if necessary.
+	 * @param commandLineArgs The command line arguments to be converted
+	 * @return list of command line args where app args are now associated with the comosed-task-app arguments and all other remain unchanged.
```

Comment:
**[TinyNit]** this was a bit confusing for me the 1st few reads. Another way to state this would be 
```

/**
 * Converts command lines args into a format acceptable for CTR. 
 * <p>The input args are copied and entries that begin with {@code 'app.'} 
 * are  replaced with a {@code 'composed-task-app-arguments.'} 
 * prefixed entry. The transformed arg will also be converted to Base64 
 * if necessary (eg. when it has an {@code =} sign in the value).
 * 
 * @param commandLineArgs The command line arguments to be converted
 * @return list of converted command line arguments
 */

```
which I find a little easier to follow. Wdyt?